src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/service/PrisonApiService.kt

```diff
@@ -15,11 +15,12 @@ class PrisonApiService(
     return if (useClientCredentials) prisonWebClientClientCredentials else prisonWebClient
   }
 
-  suspend fun getNonAssociationDetails(prisonerNumber: String, useClientCredentials: Boolean = false): NonAssociationDetails {
+  fun getNonAssociationDetails(prisonerNumber: String, useClientCredentials: Boolean = false): NonAssociationDetails {
     return getClient(useClientCredentials)
       .get()
       .uri("/api/offenders/$prisonerNumber/non-association-details")
       .retrieve()
-      .awaitBody()
+      .bodyToMono(object : ParameterizedTypeReference<NonAssociationDetails>() {})
```

Comment:
What is this doing 😬?

Is this equivalent to `.bodyToMono(NonAssociationDetails::class.java)`? That seems to compile and work 🤔 (and it looks slightly less weird)